ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/upnp/DefaultRegistryListener.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.p2p.upnp;
+
+import org.jupnp.model.meta.LocalDevice;
+import org.jupnp.model.meta.RemoteDevice;
+import org.jupnp.registry.Registry;
+import org.jupnp.registry.RegistryListener;
+
+class DefaultRegistryListener implements RegistryListener {
```

Comment:
Good question, @shemnon can you elaborate on this?

Our custom `DefaultRegistryListener` is different than the official one. For example:

https://github.com/jupnp/jupnp/blob/ca13ee84ae4a03be93c6bb109b27668d5406c1c5/bundles/org.jupnp/src/main/java/org/jupnp/registry/DefaultRegistryListener.java#L48

Our `remoteDeviceAdded()` doesn't chain to a more generic `deviceAdded()`. I believe this was so that we could stick with `RemoteDevice` instead of a more generic `Device`.

Perhaps we rename and/or document its purpose?

Reply:
Default registry listener brings in a boatload of unneeded dependencies and brought in yet another HTTP stack, we don't need that bloat and the JVM would slow down from yet another HTTP stack.  The DefaultRegistryListener from jupnp was mostly there to show how it would work with a set of libraries they brought in.  It also didn't work out fo the box for us.

A rename may remove that confusion.  Perhaps `LightRegistryListener` or `PantheonUpnpRegistryListener`

Reply:
Done - fde891365a1b504512424b92b17077a00a407305